onnx_caffe2/backend.py

```diff
@@ -586,6 +588,45 @@ def prepare(cls, model, device='CPU', **kwargs):
         retval = Caffe2Rep(predict_net, ws, uninitialized)
         return retval
 
+    @classmethod
+    def _caff2_op_type(cls, onnx_op_type, opset_version):
+        broken_version = cls._broken_operators.get(onnx_op_type, float('Inf'))
+        if broken_version <= opset_version:
+            raise ValueError(
+                "Don't know how to translate op {} in ONNX operator set v{} \
+                (I only support prior to v{})".format(
+                    onnx_op_type, opset_version, broken_version))
+
+        c2_op_type = cls._renamed_operators.get(onnx_op_type, onnx_op_type)
+        if not core.IsOperator(c2_op_type):
+            raise ValueError(
+                "Don't know how to translate op {}".format(onnx_op_type))
+        return c2_op_type
+
+    @classmethod
+    def validate(cls, onnx_node, opset_version):
+        onnx_op_type = onnx_node.op_type
+        c2_op_type = cls._caff2_op_type(onnx_op_type, opset_version)
+        c2_schema = OpSchema.get(c2_op_type)
+
+        num_args = len(onnx_node.attrs)
```

Comment:
It's not ok to do strict comparison on num_args since args can be optional.
Actually we want to check whether all the arguments in the resulted caffe2 operator are in the list of arguments declared in caffe2 schema (they are accessible through the `args` property of the `OpSchema` class in caffe2 python bindings.

Reply:
exactly what Junjie said. You basically want to catch a case when some argument (like `axis` here) is not supported by the converter script and gets silently swallowed. You can check that all args are present in the schema and also probably do some special care about renamed args.